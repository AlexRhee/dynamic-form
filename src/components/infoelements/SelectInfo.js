import React from 'react'

//Element for the select info (post submission)
const SelectInfo = (props) => {
    console.log(props)
    return (
        <div data-testid="selectinfo-1"  className="input-wrapper">
            <label>{props.field.name}:</label>
            <div className="inner-wrapper">
                <select disabled className="form-select" aria-label="Default select example">
                    <option >{props.field.value}</option>
                </select>
            </div>
        </div>
    )
}

export default SelectInfo;