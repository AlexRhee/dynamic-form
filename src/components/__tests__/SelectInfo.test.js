import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SelectInfo from '../infoelements/SelectInfo';

afterEach(() => {
    cleanup();
})

test('should render SelectInfo component with prefilled input information, and be disabled', () => {

    const field =         {
        "id": "jobTitle",
        "name": "Job Title",
        "options": [
            "Engineer - Lead",
            "Engineer - Mid Level",
            "Engineer - Junior",
            "Engineer - Front End Focused",
            "Engineer - Backend Focused",
            "Engineer - Full Stack"
        ],
        "placeholder": "Please select job title",
        "type": "select",
        "ind": 8,
        "value": "Engineer - Mid Level"
    }

    render(<SelectInfo field={field} />);
    const selectinfoElement = screen.getByTestId('selectinfo-1');
    expect(selectinfoElement).toBeInTheDocument();
    expect(selectinfoElement).toHaveTextContent('Job Title');
    expect(selectinfoElement).toHaveTextContent('Engineer - Mid Level');
    expect(selectinfoElement).toBeDisabled;
});

test('matches snapshot', () => {

    const field =         {
        "id": "jobTitle",
        "name": "Job Title",
        "options": [
            "Engineer - Lead",
            "Engineer - Mid Level",
            "Engineer - Junior",
            "Engineer - Front End Focused",
            "Engineer - Backend Focused",
            "Engineer - Full Stack"
        ],
        "placeholder": "Please select job title",
        "type": "select",
        "ind": 8,
        "value": "Engineer - Mid Level"
    }

    const tree = renderer.create(<SelectInfo field={field} />).toJSON();
    expect(tree).toMatchSnapshot();
})