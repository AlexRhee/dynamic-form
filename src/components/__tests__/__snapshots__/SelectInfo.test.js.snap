// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="input-wrapper"
  data-testid="selectinfo-1"
>
  <label>
    Job Title
    :
  </label>
  <div
    className="inner-wrapper"
  >
    <select
      aria-label="Default select example"
      className="form-select"
      disabled={true}
      value="Engineer - Mid Level"
    >
      <option>
        Please select job title
      </option>
      <option>
        Engineer - Lead
      </option>
      <option>
        Engineer - Mid Level
      </option>
      <option>
        Engineer - Junior
      </option>
      <option>
        Engineer - Front End Focused
      </option>
      <option>
        Engineer - Backend Focused
      </option>
      <option>
        Engineer - Full Stack
      </option>
    </select>
  </div>
</div>
`;
